### ISSUE TITLE
DocTag's - document strings tied to specific file locations

### ISSUE THREAD

**Issue by tshort** [MichaelHatherly/Docile.jl#59]
It'd be great to have something like the following:

```julia
@doctag """
# My header
The following code does x,y,z...
"""
```

That would generate an entry of type `SomeRandomDocTag` that inherits from `DocTag`. The main use is to insert headers, front matter, or back matter in documentation. The key is that the entry has the file location, so you can sort by DocTag or position to get the information where you want. When filtering/sorting the end result, you can treat the DocTags specially.

I kludged up something similar in Sims.jl, where the entry looked like:

```julia
@doc """
# Basics
""" -> type DocElBasics <: DocTag end
```

That works, but it's visually clumsy and error prone (you have to make the `DocElBasics` unique). See [here](https://github.com/tshort/Sims.jl/blob/mkdocs/lib/electrical.jl#L6-L54) for an example used for front matter. DocTags are also used as headers in that file. See [here](https://tshort.github.io/Sims.jl/lib/electrical/) for the end result.

I tried and failed to make my own `@doctag` that uses and expression based on `@doc`, but my macro skills weren't up to it. I could create the entry, but the line location was at the point of the macro definition, not where the documentation string is.

**Comment by MichaelHatherly** [comment]
Ok, yes this makes a lot more sense now :)

I like the idea &mdash; kind of heading into literate programming style, which is fine by me.

Nicer syntax would be best. `@doctag` syntax might work nicely and I'm sure we could get the correct file/line info needed.

I'm not sure of the need to be able to search through the content in the repl, though having not tried I couldn't really say either way. Thus it might be best to separate this feature from the `entries` field in `Metadata` and rather store them in the `data` field `Dict` with key `:doctags` or something like that. Then Lexicon could gather doctags from there rather than sifting through all the `entries`.

**Comment by MichaelHatherly** [comment]
Closed by 1661d16c31150878b3cdc0821053020ce96055fd.
